draft-ietf-mls-protocol.md

```diff
@@ -662,17 +662,113 @@ Cryptographic algorithms are indicated using the following types:
 
 ~~~~~
 enum {
+    /* Elliptic Curve Groups (ECDHE) */
+    secp256r1(0x0017), secp384r1(0x0018), secp521r1(0x0019),
+    x25519(0x001D), x448(0x001E),
+
+    /* Finite Field Groups (DHE) */
+    ffdhe2048(0x0100), ffdhe3072(0x0101), ffdhe4096(0x0102),
+    ffdhe6144(0x0103), ffdhe8192(0x0104),
+
+    /* Reserved */
+    (0xFFFF)
+} NamedGroup;
+
+struct {
+    NamedGroup named_group_list<2..2^16-1>;
+} NamedGroupList;
+~~~~~
+
+Elliptic Curve Groups (ECDHE)
+: Indicates support for the corresponding named curve, defined
+  either in FIPS 186-4 {{DSS}} or in {{RFC7748}}.
+
+Finite Field Groups (FFDHE)
+: Indicates support of the corresponding finite field
+  group, defined in {{RFC7919}}.
+
+~~~~~
+enum {
+    /* ECDSA algorithms */
     ecdsa_secp256r1_sha256(0x0403),
+    ecdsa_secp384r1_sha384(0x0503),
+    ecdsa_secp521r1_sha512(0x0603),
+
+    /* EdDSA algorithms */
     ed25519(0x0807),
+    ed448(0x0808),
+
+    /* RSASSA-PSS algorithms with public key OID RSASSA-PSS */
+    rsa_pss_pss_sha256(0x0809),
+    rsa_pss_pss_sha384(0x080a),
+    rsa_pss_pss_sha512(0x080b),
+
+    /* Reserved */
     (0xFFFF)
 } SignatureScheme;
 
-enum {
-    P256_SHA256_AES128GCM(0x0000),
-    X25519_SHA256_AES128GCM(0x0001),
-    (0xFFFF)
-} CipherSuite;
-~~~~~
+struct {
+    SignatureScheme supported_signature_algorithms<2..2^16-2>;
+} SignatureSchemeList;
+~~~~~
+
+ECDSA algorithms
+: Indicates a signature algorithm using ECDSA {{ECDSA}}, the corresponding
+  curve as defined in ANSI X9.62 {{X962}} and FIPS 186-4 {{DSS}}, and the
+  corresponding hash algorithm as defined in {{!SHS}}. The signature is
+  represented as a DER-encoded {{X690}} ECDSA-Sig-Value structure.
+
+EdDSA algorithms
+: Indicates a signature algorithm using EdDSA as defined in
+  {{RFC8032}} or its successors. Note that these correspond to the
+  "PureEdDSA" algorithms and not the "prehash" variants.
+
+RSASSA-PSS PSS algorithms
```

Comment:
Let's not do RSA-PSS.  I don't think anyone is asking for it.

Reply:
I actually agree with this and will remove it. I liked the idea of aligning on TLS but randomness issues related to this are not something we want.